src/rmem.c

```diff
@@ -0,0 +1,470 @@
+#include "rmem.h"
+
+
+typedef uintptr_t ptrcmp;
+
+static size_t _AlignSize(const size_t size, const size_t align)
+{
+    return (size + (align-1)) & -align;
+}
+
+
+/************* Memory Pool *************/
+
+static void _RemoveNode(struct MemNode **const node)
+{
+    ((*node)->prev != NULL)? ((*node)->prev->next = (*node)->next) : (*node = (*node)->next);
+    ((*node)->next != NULL)? ((*node)->next->prev = (*node)->prev) : (*node = (*node)->prev);
+}
+
+struct MemPool MemPool_Create(const size_t size)
+{
+    struct MemPool mempool = {0};
+    if( size==0UL )
+        return mempool;
+    else {
+        // align the mempool size to at least the size of an alloc node.
+        mempool.stack.size = _AlignSize(size, sizeof(struct MemNode));
+        mempool.stack.mem = malloc(1 + mempool.stack.size*sizeof *mempool.stack.mem);
+        if( mempool.stack.mem==NULL ) {
+            mempool.stack.size = 0UL;
+            return mempool;
+        } else {
+            mempool.stack.base = mempool.stack.mem + mempool.stack.size;
+            return mempool;
+        }
+    }
+}
+
+struct MemPool MemPool_FromBuffer(const size_t size, void *buf)
+{
+    struct MemPool mempool = {0};
+    if( size==0UL || buf==NULL || size<=sizeof(struct MemNode) )
+        return mempool;
+    else {
+        mempool.stack.size = size;
+        mempool.stack.mem = buf;
+        mempool.stack.base = mempool.stack.mem + mempool.stack.size;
+        return mempool;
+    }
+}
+
+void MemPool_Destroy(struct MemPool *const mempool)
+{
+    if( mempool==NULL || mempool->stack.mem==NULL )
+        return;
+    else {
+        free(mempool->stack.mem);
+        *mempool = (struct MemPool){0};
+    }
+}
+
+void *MemPool_Alloc(struct MemPool *const mempool, const size_t size)
+{
+    if( mempool==NULL || size==0UL || size > mempool->stack.size )
+        return NULL;
+    else {
+        struct MemNode *new_mem = NULL;
+        const size_t ALLOC_SIZE = size + sizeof *new_mem;
```

Comment:
keep ALL_CAPS identifier usage for macros.

Reply:
Ummm, the coding conventions specifically said ALL_CAPS for macros AND constants...

https://github.com/raysan5/raylib/wiki/raylib-coding-conventions

```
Code element 	Convention 	Example
Macros 	ALL_CAPS 	#define MIN(a,b) (((a)<(b))?(a):(b))
Defines 	ALL_CAPS 	#define PLATFORM_DESKTOP
Constants 	ALL_CAPS 	const int NUMBER = 8
```

Reply:
Didn't know about that one tbh. Still I don't think all caps is appropriate here, it's `const`, but it's not constant (it's variable differed depending on `size`).